cognite/client/_api/iam.py

```diff
@@ -597,11 +600,21 @@ def revoke(self, id: int | Sequence[int]) -> Session | SessionList:
         Returns:
             Session | SessionList: List of revoked sessions. If the user does not have the sessionsAcl:LIST capability, then only the session IDs will be present in the response.
         """
-        identifiers = IdentifierSequence.load(ids=id, external_ids=None)
-        items = {"items": identifiers.as_dicts()}
 
-        result = SessionList._load(self._post(self._RESOURCE_PATH + "/revoke", items).json()["items"])
-        return result[0] if isinstance(id, int) else result
+        ident_sequence = IdentifierSequence.load(ids=id, external_ids=None)
+
+        revoked_sessions_res = cast(
+            list,
+            self._delete_multiple(
+                identifiers=ident_sequence,
+                wrap_ids=True,
+                returns_items=True,
+                delete_endpoint="/revoke",
+            ),
+        )
+
+        revoked_sessions = SessionList._load(revoked_sessions_res)
+        return revoked_sessions[0] if ident_sequence.is_singleton() else revoked_sessions
```

Comment:
Does this need an "and" check or will revoke fail if the session doesn't exist (so that revoked_sessions can never be an empty list)?

Reply:
revoke fails with `CogniteAPIError: Items in request were not found | code: 400` when session id doesn't exit